js/components/admin-page-settings.js

```diff
@@ -64,6 +64,24 @@ const AdminPageSettings = function() {
 					metaKey="menu_position"
 				/>
 			</PanelRow>
+			<PanelRow>
+				<div className="hello-admin-select-multiple">
+					<MetaSelectControl
+						label={ __( 'User Roles', 'hello-admin' ) }
+						metaKey="user_roles"
+						help="Choose which user roles should see this admin page"
+						multiple={ true }
+						options={ [
+							{ value: null, label: __( 'Select User Roles', 'hello-admin' ), disabled: true },
+							{ value: 'administrator', label: __( 'Administrator', 'hello-admin' ) },
+							{ value: 'editor', label: __( 'Editor', 'hello-admin' ) },
+							{ value: 'author', label: __( 'Author', 'hello-admin' ) },
+							{ value: 'contributor', label: __( 'Contributor', 'hello-admin' ) },
+							{ value: 'subscriber', label: __( 'Subscriber', 'hello-admin' ) },
```

Comment:
It would be good if we could pull these dynamically. It's possible to add custom user roles.

Reply:
Happy to just create a new issue for this though, and do it later.

Couple of options on how it could be achieved:

- Localise the script with known user roles (seems like the simplest)
- Hope there's a user roles REST endpoint built into WP and use that
- Create our own REST endpoint to retrieve the user roles

Reply:
This sounds good and found the simple method get_editable_roles() to use but now I am having an issue with the localization method. Should be simple but I must be doing the registration wrong. The variable is undefined when trying to access it in the js file and in the console in the admin page.
`add_action( 'admin-init', [ $this, 'localize_user_roles' ], 100 );
public function localize_user_roles() {
		$version = hello_admin_version();

		wp_register_script(
			'admin-page-settings-js',
			plugins_url( '/js/components/admin-page-settings.js', dirname( __FILE__ ) ),
			[ 'wp-data', 'wp-i18n', 'wp-element', 'wp-editPost', 'wp-components', 'wp-apiFetch' ],
			$version,
			true
		);

		wp_localize_script(
			'admin-page-settings-js',
			'helloAdminUserRoles',
			get_editable_roles()
		);
	}`

Can you see what I am doing wrong? I have basically copied this method from how we did it in Hello Charts. That variable still appears in the console but can't find helloAdminUserRoles.

Reply:
What happens when you move the localization to after the script is _enqueued_ instead of after it's registered?

Reply:
Like this?

`
wp_enqueue_script(
			'admin-page-settings-js',
			plugins_url( '/js/components/admin-page-settings.js', dirname( __FILE__ ) ),
			[ 'wp-data', 'wp-i18n', 'wp-element', 'wp-editPost', 'wp-components', 'wp-apiFetch' ],
			$version,
			true
		);

		wp_localize_script(
			'admin-page-settings-js',
			'helloAdminUserRoles',
			get_editable_roles()
		);
`
I did try to enqueue but it was the same result. Still undefined.
![image](https://user-images.githubusercontent.com/42702782/144979472-61d4d4f6-a38a-47ae-9913-6f005bc1c3ae.png)